docs/dev/testing_github_app_auth.md

```diff
@@ -0,0 +1,34 @@
+# Testing GitHub app auth
+
+## Step 1: Create and install a dummy GitHub app for testing with
+
+Go to https://github.com/settings/apps/new
+
+1. Enter a name for the app (needs to be unique across GitHub).
+2. Set the required `homepage URL` field (can be any valid URL).
+3. Under `Webhook`, uncheck the `Active` checkbox.
+4. Click on `Repository permissions` under `Permissions`, and set `Contents` to `Read-only`
+5. Click on `Create GitHub App` at the bottom of the page.
+6. You should be navigated to a new page with a `Registration successful. You must generate a private key in order to install your GitHub App.` message. Click on the `generate a private key` link, and then the `Generate a private key` button, and save it somewhere; it will be used to test the app authentication.
+7. Click on the `Install App` tab on the left, and then click on `Install` on the right.
+8. Select `Only select repositories`, and pick any private repository that contains a "LICENSE" file (may need to be created beforehand).
+
+## Step 2: Export the necessary environment variables
+
+The following environment variables are *required* to run the git-sync github app auth test:
+- `GITHUB_APP_PRIVATE_KEY`
```

Comment:
These env variables need to have the `GITSYNC_` prefix.

Reply:
These ones don't correspond to the actual application env vars though, they're the values that get passed in via flags to the `GIT_SYNC` function in the test